go-packages/meep-model/model.go

```diff
@@ -943,6 +957,737 @@ func (m *Model) GetNodeContext(name string) (ctx interface{}) {
+			}
+		}
+	case "zone":
+		if filter.ZoneName != "" {
+			n = m.nodeMap.FindByName(filter.ZoneName)
+			if n != nil {
+				nList = append(nList, n)
+			}
+		} else {
+			nList = m.nodeMap.FindAllByType(NodeTypeZone)
+		}
+		//only one type of zone... so no need to check for a different filter
+	case "networkLocation":
+		if filter.NetworkLocationName != "" {
+			if filter.NetworkLocationType != "" {
+				n = m.nodeMap.FindByType(filter.NetworkLocationName, filter.NetworkLocationType)
+				if n != nil {
+					nList = append(nList, n)
+				}
+			} else {
+				n = m.nodeMap.FindByName(filter.NetworkLocationName)
+				if n != nil {
+					nList = append(nList, n)
+				}
+			}
+		} else {
+			if filter.NetworkLocationType != "" {
+				nList = m.nodeMap.FindAllByType(filter.NetworkLocationType)
+			} else {
+				nList = m.nodeMap.FindAllByType(NodeTypePoa)
+				nListOther := m.nodeMap.FindAllByType(NodeTypePoa4G)
+				nList = append(nList, nListOther...)
+				nListOther = m.nodeMap.FindAllByType(NodeTypePoa5G)
+				nList = append(nList, nListOther...)
+				nListOther = m.nodeMap.FindAllByType(NodeTypePoaWifi)
+				nList = append(nList, nListOther...)
+			}
+		}
+	case "physicalLocation":
+		if filter.PhysicalLocationName != "" {
+			if filter.PhysicalLocationType != "" {
+				n = m.nodeMap.FindByType(filter.PhysicalLocationName, filter.PhysicalLocationType)
+				if n != nil {
+					nList = append(nList, n)
+				}
+			} else {
+				n = m.nodeMap.FindByName(filter.PhysicalLocationName)
+				if n != nil {
+					nList = append(nList, n)
+				}
+			}
+		} else {
+			if filter.PhysicalLocationType != "" {
+				nList = m.nodeMap.FindAllByType(filter.PhysicalLocationType)
+			} else {
+				nList = m.nodeMap.FindAllByType(NodeTypeCloud)
+				nListOther := m.nodeMap.FindAllByType(NodeTypeEdge)
+				nList = append(nList, nListOther...)
+				nListOther = m.nodeMap.FindAllByType(NodeTypeFog)
+				nList = append(nList, nListOther...)
+				nListOther = m.nodeMap.FindAllByType(NodeTypeUE)
+				nList = append(nList, nListOther...)
+			}
+		}
+	case "process":
+		if filter.ProcessName != "" {
+			if filter.ProcessType != "" {
+				n = m.nodeMap.FindByType(filter.ProcessName, filter.ProcessType)
+				if n != nil {
+					nList = append(nList, n)
+				}
+			} else {
+				n = m.nodeMap.FindByName(filter.ProcessName)
+				if n != nil {
+					nList = append(nList, n)
+				}
+			}
+		} else {
+			if filter.ProcessType != "" {
+				nList = m.nodeMap.FindAllByType(filter.ProcessType)
+			} else {
+				nList = m.nodeMap.FindAllByType(NodeTypeEdgeApp)
+				nListOther := m.nodeMap.FindAllByType(NodeTypeUEApp)
+				nList = append(nList, nListOther...)
+			}
+		}
+	default:
+	}
+	return nList
+}
+
+// GetNodesByParentLevelFilter - Get list of nodes that match the filter values for all parents above the current hierarchy level provided
+//              Returned value is of type []*Node
+func (m *Model) GetNodesByParentLevelFilter(level string, filter *NodeFindFilter) []*Node {
```

Comment:
Try to avoid using levels unless necessary.

Reply:
code restructured...